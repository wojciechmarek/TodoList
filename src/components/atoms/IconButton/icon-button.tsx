import { IconButtonProps } from './icon-button.interface';
import { IconButtonStyle } from './icon-button.styled';

export const IconButton = (props: IconButtonProps) => {
  const { children, ...params } = props;

  return <IconButtonStyle {...params}>{children}</IconButtonStyle>;
};
