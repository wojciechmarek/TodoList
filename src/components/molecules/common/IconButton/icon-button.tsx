import { Icon } from '../../../atoms';
import { IconButtonProps } from './icon-button.interface';
import { IconButtonStyle } from './icon-button.styled';

export const IconButton = (props: IconButtonProps) => {
  const { icon, ...params } = props;

  return (
    <IconButtonStyle {...params}>
      <Icon>{icon}</Icon>
    </IconButtonStyle>
  );
};
