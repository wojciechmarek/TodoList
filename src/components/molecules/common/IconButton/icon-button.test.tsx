import { configure, render } from '@testing-library/react';
import { IconButton } from './icon-button';
import { IconButtonProps } from './icon-button.interface';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';

describe('IconButton', () => {
  let iconButtonProps: IconButtonProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    iconButtonProps = {
      icon: <img src="https://via.placeholder.com/150" alt="icon" />,
      onClick: jest.fn(),
    };
  });

  it('should render', () => {
    const { baseElement } = render(<IconButton {...iconButtonProps} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(
      <IconButton {...iconButtonProps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when the icon button is rendered', () => {
    it('should render the icon', () => {
      const { getByRole } = render(<IconButton {...iconButtonProps} />);
      const image = getByRole('img', { name: /icon/i });
      expect(image).toBeTruthy();
    });
  });

  describe('when the icon button is clicked', () => {
    it('should call the onClick function', () => {
      const { getByRole } = render(<IconButton {...iconButtonProps} />);
      const button = getByRole('button', { name: /icon/i });
      button.click();
      expect(iconButtonProps.onClick).toHaveBeenCalled();
    });

    it('should not call the onClick function when the button is disabled', () => {
      const { getByRole } = render(
        <IconButton {...iconButtonProps} disabled />
      );
      const button = getByRole('button', { name: /icon/i });
      button.click();
      expect(iconButtonProps.onClick).not.toHaveBeenCalled();
    });
  });
});
