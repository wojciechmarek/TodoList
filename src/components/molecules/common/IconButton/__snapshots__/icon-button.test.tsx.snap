// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match the snapshot 1`] = `
.emotion-0 {
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: none;
}

.emotion-0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.emotion-2 {
  height: calc(100% * 0.618);
  width: calc(100% * 0.618);
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <img
        alt="icon"
        src="https://via.placeholder.com/150"
      />
    </div>
  </div>
</button>
`;
